programs/nft-voter/src/instructions/update_voter_weight_record.rs

```diff
@@ -0,0 +1,71 @@
+use crate::{
+    error::NftVoterError
+};
+use crate::state::*;
+use anchor_lang::prelude::*;
+use itertools::Itertools;
+
+
+/// Updates VoterWeightRecord to evaluate governance power for non voting use cases: CreateProposal, CreateGovernance etc...
+/// This instruction updates VoterWeightRecord which is valid for the current Slot and the given target action only
+/// and hance the instruction has to be executed inside the same transaction as the corresponding spl-gov instruction
+#[derive(Accounts)]
+#[instruction(voter_weight_action:VoterWeightAction)]
+pub struct UpdateVoterWeightRecord<'info> {
+    /// The NFT voting Registrar
+    pub registrar: Account<'info, Registrar>,
+
+    #[account(
+        mut,
+        constraint = voter_weight_record.realm == registrar.realm 
+        @ NftVoterError::InvalidVoterWeightRecordRealm,
+
+        constraint = voter_weight_record.governing_token_mint == registrar.governing_token_mint
+        @ NftVoterError::InvalidVoterWeightRecordMint,
+    )]
+    pub voter_weight_record: Account<'info, VoterWeightRecord>,
+}
+
+pub fn update_voter_weight_record(
+    ctx: Context<UpdateVoterWeightRecord>,
+    voter_weight_action: VoterWeightAction,
+) -> Result<()> {
+
+    let registrar = &ctx.accounts.registrar;
+    let governing_token_owner = &ctx.accounts.voter_weight_record.governing_token_owner;
+
+    // CastVote can't be evaluated using this instruction 
+    require!(
+        voter_weight_action != VoterWeightAction::CastVote,
+        NftVoterError::CastVoteIsNotAllowed
+    );
+
+    let mut voter_weight = 0u64;
+
+    // Ensure all nfts are unique
+    let mut unique_nft_mints = vec![];
+
+    for (nft_info, nft_metadata_info) in ctx.remaining_accounts.iter().tuples() {
```

Comment:
I'm sure you're aware of this, but because you can only get ~32 accounts in an transaction, there's a limit of ~15 NFTs per voter, with no way of aggregating over multiple transactions.  Is that going to be ok for users?

Reply:
Yes, I'm aware of that limitation. I'm planning to support multi tx voter weight update in future versions for larger amount of NFTs